core/unit_test/incremental_testing/Test02b_MemorySpace.hpp

```diff
@@ -47,21 +47,22 @@
 #include <type_traits>
 #include <gtest/gtest.h>
 
-
 namespace Test {
 
-// Test construction and assignment
+// Unit test for kokkos_free
+// We constantly allocate and de-allocate memory.
+// If the kokkos_free does not free the memory, we will exceed the available
```

Comment:
This is only true if failing to deallocate would result in allocating more than the entire virtual memory space.  Linux overcommit means that if you don't actually touch the allocations, then you can allocate way more than the physical memory capacity.